src/assets/manifest.ts

```diff
@@ -54,3 +37,27 @@ export async function writeWebManifest(
     "utf-8",
   );
 }
+
+export async function overrideWebManifestIcons(manifestFile: string) {
+  const manifest = await readWebManifestFile(manifestFile);
+
+  return !!manifest?.manifest && !("icons" in manifest.manifest);
```

Comment:
what if the icons config is changed? for instance by default we're on minimal-2023, but what if we change it in the config, this would not change it, right? So it'd be a bug.

Reply:
The template by default doesn't have icons entry, so it should be fine.

We need to cover these cases, since the override icons flag is enabled by default (you don't know how the user will use it):
- the user can use the pwa assets from CLI and update the manifest with CLI suggestions (for example adding the icons)
- the user can use custom pwa assets config file: shouldn't be a problem if not adding icons to the manifest
- the user can use the default behavior: shouldn't be a problem

If the user using the first approach, then it should update the manifest properly after re-runnning the CLI, maybe also requires a server restart

Reply:
I know, but imagine we inject some icons by preset 2023, and now the user changes the preset to something else? how this is going to updates the entry?

Reply:
check assets plugin, if the preset changes then vite will restart dev server and the new one will be loaded, if using external pwa assets config file, assets plugin will reload the configuration and invalidate the cache entries, forcing a page reload.